assets/js/modules/analytics-4/components/setup/SetupEnhancedMeasurementSwitch.js

```diff
@@ -109,30 +113,15 @@ export default function SetupEnhancedMeasurementSwitch() {
 		);
 	} );
 
-	const isEnhancedMeasurementEnabled = useSelect( ( select ) =>
-		select( CORE_FORMS ).getValue(
-			ENHANCED_MEASUREMENT_FORM,
-			ENHANCED_MEASUREMENT_ENABLED
-		)
-	);
-
 	const { setValues } = useDispatch( CORE_FORMS );
 
-	// If `isEnhancedMeasurementEnabled` is already defined in the first render, it means we're rendering this component
-	// after the user has actively selected the setting, in which case we don't want to override the setting to `true` in
-	// the `useEffect()` unless the account, property, or web data stream selection is subsequently changed.
-	const skipEffect = useRef( isEnhancedMeasurementEnabled !== undefined );
-
-	useEffect( () => {
```

Comment:
This hook was always overriding the EM value, as it listens to the property/webdata stream values change in datastore, and when we create a new one, on OAuth redirect, value(s) would change from `PROPERTY_CREATE`/`WEBDATASTREAM_CREATE` to the actual proeprty/web data stream ID which triggers this hook and always sets EM to on. Doesn't seem possible to differentiate this in the hook, so I simplified the logic, so EM value is reset when actual account/property/webdata stream dropdown is changed by the user